include/SFML/Window/Keyboard.hpp

```diff
@@ -153,6 +158,158 @@ class SFML_WINDOW_API Keyboard
+        ScanComma,             ///< Keyboard , and < key
+        ScanPeriod,            ///< Keyboard . and > key
+        ScanForwardSlash,      ///< Keyboard / and ? key
+        ScanF1,                ///< Keyboard F1 key
+        ScanF2,                ///< Keyboard F2 key
+        ScanF3,                ///< Keyboard F3 key
+        ScanF4,                ///< Keyboard F4 key
+        ScanF5,                ///< Keyboard F5 key
+        ScanF6,                ///< Keyboard F6 key
+        ScanF7,                ///< Keyboard F7 key
+        ScanF8,                ///< Keyboard F8 key
+        ScanF9,                ///< Keyboard F9 key
+        ScanF10,               ///< Keyboard F10 key
+        ScanF11,               ///< Keyboard F11 key
+        ScanF12,               ///< Keyboard F12 key
+        ScanF13,               ///< Keyboard F13 key
+        ScanF14,               ///< Keyboard F14 key
+        ScanF15,               ///< Keyboard F15 key
+        ScanCapsLock,          ///< Keyboard Caps Lock key
+        ScanPrintScreen,       ///< Keyboard Print Screen key
+        ScanScrollLock,        ///< Keyboard Scroll Lock key
+        ScanPause,             ///< Keyboard Pause key
+        ScanInsert,            ///< Keyboard Insert key
+        ScanHome,              ///< Keyboard Home key
+        ScanPageUp,            ///< Keyboard Page Up key
+        ScanDelete,            ///< Keyboard Delete Forward key
+        ScanEnd,               ///< Keyboard End key
+        ScanPageDown,          ///< Keyboard Page Down key
+        ScanRight,             ///< Keyboard Right Arrow key
+        ScanLeft,              ///< Keyboard Left Arrow key
+        ScanDown,              ///< Keyboard Down Arrow key
+        ScanUp,                ///< Keyboard Up Arrow key
+        ScanNumLock,           ///< Keypad Num Lock and Clear key
+        ScanDivide,            ///< Keypad / key
+        ScanMultiply,          ///< Keypad * key
+        ScanMinus,             ///< Keypad - key
+        ScanPlus,              ///< Keypad + key
+        ScanPadEquals,         ///< keypad = key, probably Mac only
+        ScanReturn,            ///< Keypad Enter (return) key
+        ScanDecimal,           ///< Keypad . and Delete key
+        ScanNumpad1,           ///< Keypad 1 and End key
+        ScanNumpad2,           ///< Keypad 2 and Down Arrow key
+        ScanNumpad3,           ///< Keypad 3 and Page Down key
+        ScanNumpad4,           ///< Keypad 4 and Left Arrow key
+        ScanNumpad5,           ///< Keypad 5 key
+        ScanNumpad6,           ///< Keypad 6 and Right Arrow key
+        ScanNumpad7,           ///< Keypad 7 and Home key
+        ScanNumpad8,           ///< Keypad 8 and Up Arrow key
+        ScanNumpad9,           ///< Keypad 9 and Page Up key
+        ScanNumpad0,           ///< Keypad 0 and Insert key
+        ScanReverseSolidus,    ///< Keyboard Non-US \ and | key
```

Comment:
How does this differ from a backslash? On my QWERTY keyboard this key is the only way to type "\"

Reply:
> On my QWERTY keyboard this key is the only way to type ""

To type quotes?

Honestly, after ready the USB HID standard, I still don't understand what this key is...

Reply:
Isn't ScanReverseSolidus just the key between the left shift and Z on the image below (there is a backslash there on my [Belgian layout](https://upload.wikimedia.org/wikipedia/commons/thumb/9/93/Belgian_keyboard_layout.png/799px-Belgian_keyboard_layout.png)), while ScanBackslash is a key found near the return key?
![US keyboard layout](https://upload.wikimedia.org/wikipedia/commons/thumb/1/17/LinuxUSKeyboard.png/800px-LinuxUSKeyboard.png)

In SDL it seems to be called SDL_SCANCODE_NONUSBACKSLASH.

Reply:
This is probably it, yes.

Reply:
As mentioned above, 'reverse solidus' literally just means 'backwards slash' so using both makes no sense at all.
In the layout image shown by texus, the key proposed (left of z) for reverse solidus doesn't even have that on it.
Would it not be more prudent to just name it "pipe" (i.e. ScanPipe) or something like that?

Reply:
@mantognini, I meant `\`. Github just removed backslash somehow.
But looking at layout that @texus has shown, it seems pretty unusual for me. Yes, ScanPipe seems less ambigious.

https://wiki.libsdl.org/SDL_Scancode
SDL named it "Backslash". Maybe it produces the same keycode in different layouts? On my laptop keyboard this button is above enter, but on the other one it's left to it, same as on layout @texus displayed.

@texus, can you show what "xev" shows when you press different buttons with "pipe" on it?

Reply:
It seems like in many cases it contains the `<` and `>` symbols, so it could be named after them, but there are keyboard (e.g. UK layout) where there is only a backslash and pipe on that key.

I'm not sure what output from xev you want exactly, on my keyboard layout the pipe key is even located at the 1 number key. Bet below are some xev outputs though.

Pressing the ScanReverseSolidus key: (first one is normal press, second one is with shift and third one is with AltGr)
```
KeyPress event, serial 37, synthetic NO, window 0x4200001,
    root 0x125, subw 0x0, time 42422739, (-320,124), root:(843,535),
    state 0x10, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XmbLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False

KeyPress event, serial 37, synthetic NO, window 0x4200001,
    root 0x125, subw 0x0, time 42425023, (-320,124), root:(843,535),
    state 0x11, keycode 94 (keysym 0x3e, greater), same_screen YES,
    XLookupString gives 1 bytes: (3e) ">"
    XmbLookupString gives 1 bytes: (3e) ">"
    XFilterEvent returns: False

KeyPress event, serial 37, synthetic NO, window 0x4200001,
    root 0x125, subw 0x0, time 42426153, (-320,124), root:(843,535),
    state 0x90, keycode 94 (keysym 0x5c, backslash), same_screen YES,
    XKeysymToKeycode returns keycode: 20
    XLookupString gives 1 bytes: (5c) "\"
    XmbLookupString gives 1 bytes: (5c) "\"
    XFilterEvent returns: False
```

I get the following when pressing the key where the backslash would be according to the layout in the image I showed in my previous comment:
```
KeyPress event, serial 34, synthetic NO, window 0x6c00001,
    root 0x125, subw 0x0, time 43321539, (-263,93), root:(919,550),
    state 0x10, keycode 51 (keysym 0xb5, mu), same_screen YES,
    XLookupString gives 2 bytes: (c2 b5) "µ"
    XmbLookupString gives 2 bytes: (c2 b5) "µ"
    XFilterEvent returns: False
```

To get the actual pipe symbol I have to press AltGr+1:
```
KeyPress event, serial 37, synthetic NO, window 0x6c00001,
    root 0x125, subw 0x0, time 43010269, (-289,117), root:(890,504),
    state 0x90, keycode 10 (keysym 0x7c, bar), same_screen YES,
    XLookupString gives 1 bytes: (7c) "|"
    XmbLookupString gives 1 bytes: (7c) "|"
    XFilterEvent returns: False
```

Reply:
Thanks! Very helpful.

Take a look at [this](http://www.charvolant.org/~doug/xkb/html/img3.png)
So, there's a BKSL key for XKB where backslash is on the QWERTY layout. And for keycode 94 the symbolic name is "LSGT" (less, greater). Which makes sense, because it produces "<" primarly.

Reply:
Btw, this button (the one which is right to the left shift) is named "SDL_SCANCODE_NONUSBACKSLASH" in SDL.

Reply:
I don't really care about the name but for consistency we should probably remain close to the HID documentation, i.e. non US backslash or reverse solidus (the latter is easier to capitalise though).

Reply:
Then I propose to leave it as is. It's a pretty weird key, so I don't think a lot of people will use it. And those who do probably won't complain much. :)

Reply:
I've chosen `ScanNonUsBackslash` for it